module/VuFind/src/VuFind/Search/ReservesHelper.php

```diff
@@ -21,59 +21,71 @@
  * <https://www.gnu.org/licenses/>.
  *
  * @category VuFind
- * @package  Controller_Plugins
+ * @package  Search
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
 
-namespace VuFind\Controller\Plugin;
+namespace VuFind\Search;
 
-use Laminas\Mvc\Controller\Plugin\AbstractPlugin;
+use VuFind\ILS\Connection;
 use VuFindSearch\Command\RetrieveCommand;
 use VuFindSearch\Service;
 
 /**
- * Action helper to perform reserves-related actions
+ * Helper to perform reserves-related actions
  *
  * @category VuFind
- * @package  Controller_Plugins
+ * @package  Search
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     https://vufind.org Main Page
  */
-class Reserves extends AbstractPlugin
+class ReservesHelper
 {
     /**
      * Do we need to use the Solr index for reserves (true) or the ILS driver
      * (false)?
      *
      * @var bool
      */
-    protected $useIndex;
+    protected bool $useIndex;
 
     /**
      * Search service
      *
-     * @var Service
+     * @var ?Service
      */
-    protected $searchService;
+    protected ?Service $searchService;
+
+    /**
+     * ILS connection
+     *
+     * @var Connection
+     */
+    protected Connection $catalog;
 
     /**
      * Constructor
      *
-     * @param bool     $useIndex      Do we need to use the Solr index for reserves
+     * @param bool       $useIndex      Do we need to use the Solr index for reserves
      * (true) or the ILS driver (false)?
-     * @param ?Service $searchService Search service (only required when $useIndex
+     * @param ?Service   $searchService Search service (only required when $useIndex
      * is true).
+     * @param Connection $catalog       ILS connection
      */
-    public function __construct($useIndex = false, ?Service $searchService = null)
-    {
+    public function __construct(
+        bool $useIndex,
+        ?Service $searchService,
+        Connection $catalog
```

Comment:
You could use constructor property promotion here to simplify things slightly.


```suggestion
        protected ?Service $searchService,
        protected Connection $catalog
```

(And you'll have to delete the property definitions and constructor assignments as well if you accept this suggestion).

Reply:
I guess you can use constructor property promotion for all three parameters, actually -- we just need to be sure to retain the validation logic.